### ISSUE TITLE
Plugin added synchronously after init() is silently dropped

### ISSUE THREAD

**Issue by mboff-ab** [amplitude/Amplitude-TypeScript#1867]
## Expected Behavior

According to the [plugin documentation](https://amplitude.com/docs/sdks/analytics/node/node-js-sdk#plugin-examples), calling `add()` after `init()` should register the plugin normally. `setup()` should be called during initialization and `execute()` should run for every subsequent event.

## Current Behavior

If you call `add()` synchronously right after `init()`, the plugin is silently dropped. No error is thrown, nothing is logged, `setup()` is never called, and `execute()` never runs. Events go out unmodified. The call looks valid but fails silently in production with no warning.

## Steps to Reproduce

1. Install `@amplitude/analytics-node` (reproduced on `1.5.62`, latest as of this report)
2. Call `init()` then immediately `add()` a plugin in the same synchronous block
3. Call `track()` and wait for the event to flush
4. Observe that `setup()` and `execute()` on the plugin are never called

```typescript
import { init, add, track } from '@amplitude/analytics-node';
import type { EnrichmentPlugin, Event } from '@amplitude/analytics-types';

class MyPlugin implements EnrichmentPlugin {
  name = 'my-plugin';
  type = 'enrichment' as const;

  setup(): Promise<void> {
    console.log('setup called'); // never printed
    return Promise.resolve();
  }

  execute(event: Event): Promise<Event> {
    console.log('execute called'); // never printed
    event.event_properties = { ...event.event_properties, source: 'my-app' };
    return Promise.resolve(event);
  }
}

init('API_KEY');
add(new MyPlugin()); // plugin is silently dropped
track('test event'); // event_properties.source is never set
```

**Workaround:** call `add()` before `init()`. The plugin gets queued before `runQueuedFunctions` runs and is picked up correctly.

```typescript
add(new MyPlugin()); // queue first
init('API_KEY');     // plugin is registered correctly
```

A self-contained repro (no API key needed) is available [here](./repro/).

## Environment

- JS SDK Version: `1.5.62` (`@amplitude/analytics-node`)
- Installation Method: NPM
- Browser and Version: N/A - Node.js v24.12.0 (server-side SDK)

**Comment by daniel-graham-amplitude** [comment]
Thanks @mboff-ab. I can easily reproduce this. So it seems the issue is that if a function is queued up while an `init` is in progress, than that function will never be invoked. This obviously isn't good behaviour, but making fixes for these types of issues that have been around for a long time has a risk because the fix could break other functionality. I'll need to dig into this some more, but it's possible the resolution will just be to log an error message and instruct the user to do this

```
await init(...).promise;
add(...);
```

**Comment by daniel-graham-amplitude** [comment]
I just looked into it further, I think I have a fix for this: https://github.com/amplitude/Amplitude-TypeScript/pull/1873

**Comment by daniel-graham-amplitude** [comment]
This is published now. Thank you @mboff-ab !

**Comment by mboff-ab** [comment]
Thanks @daniel-graham-amplitude !
